core/src/labelled.rs

```diff
@@ -532,4 +918,77 @@ mod tests {
         let labelled = field!(name, "joe");
         assert_eq!(labelled.name, "name")
     }
+
+    #[test]
+    fn test_transmogrify_simple_identity() {
+        let one: PluckedValue<i32> = PluckedValue(1);
+        let one_again: i32 = one.transmogrify();
+        assert_eq!(one_again, 1);
+    }
+
+    #[test]
+    fn test_transmogrify_hnil_identity() {
+        let hnil_again: HNil = HNil.transmogrify();
+        assert_eq!(HNil, hnil_again);
+    }
+
+    #[test]
+    fn test_transmogrify_hcons_sculpting_super_simple() {
+        type Source = Hlist![Field<name, &'static str>, Field<age, i32>, Field<is_admin, bool>];
+        type Target = Hlist![Field<age, i32>];
+        let hcons: Source = hlist!(field!(name, "joe"), field!(age, 3), field!(is_admin, true));
+        let t_hcons: Target = hcons.transmogrify();
+        assert_eq!(t_hcons, hlist!(field!(age, 3)));
+    }
+
+    #[test]
+    fn test_transmogrify_hcons_sculpting_somewhat_simple() {
+        type Source = Hlist![Field<name, &'static str>, Field<age, i32>, Field<is_admin, bool>];
+        type Target = Hlist![Field<is_admin, bool>, Field<name, &'static str>];
+        let hcons: Source = hlist!(field!(name, "joe"), field!(age, 3), field!(is_admin, true));
+        let t_hcons: Target = hcons.transmogrify();
+        assert_eq!(t_hcons, hlist!(field!(is_admin, true), field!(name, "joe")));
+    }
+
+    #[test]
+    fn test_transmogrify_hcons_recursive_simple() {
+        type Source = Hlist![
+            Field<name,  Hlist![
+                Field<inner, f32>,
+                Field<is_admin, bool>,
+            ]>,
+            Field<age, i32>,
+            Field<is_admin, bool>];
+        type Target = Hlist![
+                Field<is_admin, bool>,
+                Field<name,  Hlist![
+                    Field<is_admin, bool>,
+                ]>,
+            ];
+        let source: Source = hlist![
+            field!(name, hlist![field!(inner, 42f32), field!(is_admin, true)]),
+            field!(age, 32),
+            field!(is_admin, true)
+        ];
+        let target: Target = source.transmogrify();
+        assert_eq!(
+            target,
+            hlist![
+                field!(is_admin, true),
+                field!(name, hlist![field!(is_admin, true)]),
+            ]
+        )
```

Comment:
Because a lot of work is put into making identity transforms work, I'd like to see tests for:

* Identity transform with labeled fields

```rust
type Source = Hlist![
    Field<name,  Hlist![
        Field<inner, f32>,
        Field<is_admin, bool>,
    ]>,
    Field<age, i32>,
    Field<is_admin, bool>];
type Target = Source;
```

* Non-identity transform where a labelled field gets an identity transform

```rust
type Source = Hlist![
    Field<name,  Hlist![
        Field<inner, f32>,
        Field<is_admin, bool>,
    ]>,
    Field<age, i32>,
    Field<is_admin, bool>];
type Source = Hlist![
    Field<age, i32>,
    Field<name,  Hlist![
        Field<inner, f32>,
        Field<is_admin, bool>,
    ]>];
```

Reply:
Yea, pretty sure at this point that the clash happens due to ambiguity of the inner fields that happen to be *both* identical **and** transmogrifiable; the two things are not mutually exclusive :( 

Not entirely sure how to get around it at this point. Any ideas?